src/ui/public/elasticsearch_errors/is_term_size_zero_error.js

```diff
@@ -1,6 +1,10 @@
 import ElasticsearchError from './elasticsearch_error';
 
 export default function isTermSizeZeroError(error) {
-  const esError = new ElasticsearchError(error);
-  return esError.hasRootCause('size must be positive, got 0');
+  try {
+    const esError = new ElasticsearchError(error);
+    return esError.hasRootCause('size must be positive, got 0');
+  } catch (err) {
+    return false;
+  }
```

Comment:
These are some great ideas! Are you suggesting doing away with shape-validation in your stateless functional solution? If not, where would an error be thrown?

Also, at the risk of going off on a tangent... I agree you shouldn't be able to instantiate a class in an unacceptable/unexpected state (who would want that? :), but I want to point out that my suggestion wasn't to do that. My suggestion was only that we could decide not to make `ElasticsearchError` responsible for validating the shape of the ES error response, and instead we could either decide not to enforce that contract or we could enforce it elsewhere. 😄